### ISSUE TITLE
Working with non-ascii URLs

### ISSUE THREAD

**Issue by vvzvlad** [html2rss/html2rss-web#525]
The application doesn't work with my address "https://снми.рф/архив/". 
If I specify it as "https://xn--h1ahcp.xn--p1ai/%D0%B0%D1%80%D1%85%D0%B8%D0%B2/", I get error "ArgumentError: malformed format string - %D", if I specify "https://снми.рф/архив/" or "https://xn--h1ahcp.xn--p1ai/архив/" I get error "URI::InvalidURIError: URI must be ascii only".

**Comment by gildesmarais** [comment]
Hi,

thank you for your report.

How did you install html2rss? 

a) Did you `gem install html2rss`?
b) Did you use bundler?

**If a:**

The latest _released_ version is really old. 

Try installing the latest version directly from git:

```
gem install specific_install
gem specific_install https://github.com/html2rss/html2rss.git master
```

**If b:**
Try adding to your `Gemfile`: 
```
gem 'html2rss', git: 'https://github.com/html2rss/html2rss'
``` 

and `bundle` afterwards. 

[Bundle docs for git support](https://bundler.io/guides/git.html)

---

Can you then try again and report back please?

**Comment by vvzvlad** [comment]
I am using docker.

```root@Farewell:~# cat /mnt/user/appdata/html2rss/feeds.yml 
stylesheets:
  - href: '/rss.xsl'
    media: 'all'
    type: 'text/xsl'
headers:
  "User-Agent": "Mozilla/5.0 (Windows NT 10.0; rv:68.0) Gecko/20100101 Firefox/68.0"
feeds:
  # your custom feeds go here:
  example:
    channel:
      url: "https://xn--h1ahcp.xn--p1ai/архив/"
    selectors:
      items:
        selector: '.news-list > li'
      title:
        selector: 'a'
      description:
        selector: 'p'```

**Comment by gildesmarais** [comment]
The fix works for:

```yml
channel:
  url: "https://снми.рф/архив/"
```

Can you update your local docker and try it again? The pipelines to update the docker image might still take a few minutes. Thank you!

**Comment by vvzvlad** [comment]
> Can you update your local docker and try it again? The pipelines to update the docker image might still take a few minutes. Thank you!

How can I verify that I really upgraded to the new version? The -v command, maybe? What version should it be?

**Comment by gildesmarais** [comment]
I'd stop the html2rss-web container and then `docker pull gilcreator/html2rss-web`

Should print this (see digest):

```
Using default tag: latest
latest: Pulling from gilcreator/html2rss-web
Digest: sha256:6020d5b1071b63aec63996bd2ea0ce92355532f98a22978b794f0aed090b77d3
```

Then, start the	container like you normally do (`docker run …`).

**Comment by vvzvlad** [comment]
Yes, it worked that way, thank you!

I should also note that for some unknown reason, the example section with the url must be in the config, otherwise health_check.txt gives a config error. Maybe that's the way it's supposed to be. I renamed my section, but until I got the example back, I was getting a config error. 

Another issue that concerns me is that the result is very much cached, because I do not see changes in the resulting feed, no matter how I do not change the config. It helped me to change the name of the feed every time I changed the config: nsmi1, nsmi2, nsmi3... and ask for a new address each time: http://192.168.88.111:3000/nsmi1.rss, http://192.168.88.111:3000/nsmi2.rss, and so on. After that, I was able to get an adequate response to the change of the config. Perhaps it makes sense to mention this in the documentation, and maybe add some sort of "nocache" flag? For example, http://IP:3000/example.rss&cache=no for debugging. 

And one last question I'd like to ask: is it possible to somehow get the content in description from the page to which the link in the news header refers? If you went to the site we were talking about (https://xn--h1ahcp.xn--p1ai/%D0%B0%D1%80%D1%85%D0%B8%D0%B2/), you saw that there is no news body there - the link is just a list of links to pages, every day - one news, one link, one text on the link page. I would really like to read these news without leaving RSS reader (for example, I can't have internet when I read news and I would like to read it from cache), but in order to do it I need parser to take content of description tag from link page. I looked through the project documentation and didn't find anything like that there. 
Maybe I wasn't looking closely?
